api/src/main/java/org/openmrs/module/fhir2/FhirTask.java

```diff
@@ -71,20 +71,47 @@
 	 * Referenced resources represented with relative resource identifier string in the format of
 	 * <ResourceName>/<ResourceId>.
 	 */
+	@ElementCollection
+	@CollectionTable(name = "fhir_task_based_on", joinColumns = @JoinColumn(name = "task_id"))
 	@Column(name = "based_on")
-	private String basedOn;
+	private Collection<String> basedOnReferences;
+	
+	/**
+	 * Referenced resources represented with relative resource identifier string in the format of
+	 * <ResourceName>/<ResourceId>.
+	 */
+	@Column(name = "for")
+	private String forReference;
```

Comment:
Might it be worth it to create our own `Reference` type that's marked as `@Embedded` or similar that can handle parsing the reference properly?